File: src/main/java/com/orange/callhistory/dao/CallEventDao.java
package com.orange.callhistory.dao;

import java.time.OffsetDateTime;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

import com.orange.callhistory.service.entity.CallEventStatus;

@Entity
public class CallEventDao {
    @Id
    private String id;

    private CallEventStatus status;

    private OffsetDateTime timestamp;

    @ManyToOne
    private CallDao callDao;

    // constructor needed by hibernate
    public CallEventDao() {
    }

    public CallEventDao(String id, CallEventStatus status, OffsetDateTime timestamp) {

        this.id = id;
        this.status = status;
        this.timestamp = timestamp;
    }


    public String getId() {
        return id;
    }

    public CallEventStatus getStatus() {
        return status;
    }

    public OffsetDateTime getTimestamp() {
        return timestamp;
    }

    void setCallDao(CallDao callDao) {
        this.callDao = callDao;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        CallEventDao that = (CallEventDao) o;

        if (id != null ? !id.equals(that.id) : that.id != null) return false;
        if (status != that.status) return false;
        return timestamp != null ? timestamp.equals(that.timestamp) : that.timestamp == null;
    }

    @Override
    public int hashCode() {
        int result = id != null ? id.hashCode() : 0;
        result = 31 * result + (status != null ? status.hashCode() : 0);
        result = 31 * result + (timestamp != null ? timestamp.hashCode() : 0);
        return result;
    }
}

File: src/main/java/com/orange/callhistory/dao/CallEventDao.kt
package com.orange.callhistory.dao

import com.orange.callhistory.service.CallEventStatus
import java.time.OffsetDateTime
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.ManyToOne

@Entity
class CallEventDao {
    @Id
    lateinit var id: String

    lateinit var status: CallEventStatus

    lateinit var timestamp: OffsetDateTime

    @ManyToOne
    private var callDao: CallDao? = null

    // constructor needed by hibernate
    constructor() {
    }

    constructor(id: String, status: CallEventStatus, timestamp: OffsetDateTime) {

        this.id = id
        this.status = status
        this.timestamp = timestamp
    }

    internal fun setCallDao(callDao: CallDao) {
        this.callDao = callDao
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val that = o as CallEventDao?

        if (if (id != null) id != that!!.id else that!!.id != null) return false
        if (status != that.status) return false
        return if (timestamp != null) timestamp == that.timestamp else that.timestamp == null
    }

    override fun hashCode(): Int {
        var result = id?.hashCode() ?: 0
        result = 31 * result + (status?.hashCode() ?: 0)
        result = 31 * result + (timestamp?.hashCode() ?: 0)
        return result
    }
}
